internal/internal_worker_test.go

```diff
@@ -1498,3 +1507,55 @@ func TestTestValidateFnFormat_Workflow(t *testing.T) {
 		})
 	}
 }
+
+func TestGetTaskAutoConfigHint(t *testing.T) {
+
+	hint := shared.AutoConfigHint{
+		EnableAutoConfig:   common.BoolPtr(true),
+		PollerWaitTimeInMs: common.Int64Ptr(100),
+	}
+
+	for _, tt := range []struct {
+		name string
+		task interface{}
+		want *shared.AutoConfigHint
+	}{
+		{
+			"decision task",
+			&workflowTask{
+				task: &shared.PollForDecisionTaskResponse{AutoConfigHint: &hint}},
+			&hint,
+		},
+		{
+			"empty decision task",
+			&workflowTask{
+				autoConfigHint: &hint},
+			&hint,
+		},
+		{
+			"activity task",
+			&activityTask{
+				task: &shared.PollForActivityTaskResponse{AutoConfigHint: &hint}},
+			&hint,
+		},
+		{
+			"empty activity task",
+			&activityTask{
+				autoConfigHint: &hint},
+			&hint,
+		},
+		{
+			"localactivity task",
+			&localActivityTask{},
+			nil,
+		},
```

Comment:
maybe just replace the one nil-test with a
```go
_, ok := &localActivityTask{}.(autoConfigHintAwareTask)
assert.False(t, ok)
```
since I that's what is actually being checked in that case, and then the others all just become an obvious "must equal tt.want"